### PR TITLE
[docs] MPS

### PR DESCRIPTION
**Pull request by stevhliu**
As a part of a larger effort to clean up the `Trainer` API docs in #27986, this PR moves the [Trainer for accelerated PyTorch training on Mac](https://huggingface.co/docs/transformers/main/en/main_classes/trainer#using-trainer-for-accelerated-pytorch-training-on-mac) section to the currently empty [Training on Specialized Hardware](https://huggingface.co/docs/transformers/main/en/perf_train_special) page.

Other updates include rewriting it a bit so it doesn't sound like it's copied directly from the blog post and removing the link to the paywalled article for setup 🙂

### CODE REVIEW

File: docs/source/en/perf_train_special.md [hunk 1]
```diff
@@ -13,12 +13,51 @@ rendered properly in your Markdown viewer.
...
-# Training on Specialized Hardware
+# PyTorch training on Apple silicon
 
-<Tip>
+Previously, training models on a Mac was limited to the CPU only. With the release of PyTorch v1.12, you can take advantage of training models with Apple's silicon GPUs for significantly faster performance and training. This is powered in PyTorch by integrating Apple's Metal Performance Shaders (MPS) as a backend. The [MPS backend](https://pytorch.org/docs/stable/notes/mps.html) implements PyTorch operations as custom Metal shaders and places these modules on a `mps` device.
 
- Note: Most of the strategies introduced in the [single GPU section](perf_train_gpu_one) (such as mixed precision training or gradient accumulation) and [multi-GPU section](perf_train_gpu_many) are generic and apply to training models in general so make sure to have a look at it before diving into this section.
+<Tip warning={true}>
+
+Some PyTorch operations are not implemented in MPS yet and will throw an error. To avoid this, you should set the environment variable `PYTORCH_ENABLE_MPS_FALLBACK=1` to use the CPU kernels instead (you'll still see a `UserWarning`).
```
Comment (amyeroberts) [inline docs/source/en/perf_train_special.md:22]: Is there a way to have trainer just use the CPU entirely and ignore the MPS backend?

File: docs/source/en/main_classes/trainer.md [hunk 2]
```diff
@@ -518,68 +518,6 @@ Pass `--fsdp "full shard"` along with following changes to be made in `--fsdp_co
...
-  --overwrite_output_dir
-```
-
-**A few caveats to be aware of**
-
-1. Some PyTorch operations have not been implemented in mps and will throw an error. 
-One way to get around that is to set the environment variable `PYTORCH_ENABLE_MPS_FALLBACK=1`, 
-which will fallback to CPU for these operations. It still throws a UserWarning however.
-2. Distributed setups `gloo` and `nccl` are not working with `mps` device. 
-This means that currently only single GPU of `mps` device type can be used.
```
Comment (amyeroberts) [inline docs/source/en/main_classes/trainer.md:576]: Is this no longer the case?

**Review by amyeroberts — COMMENTED** [review]
Very nice - thanks for reworking and tidying up!

**Review by amyeroberts — APPROVED** [review]

File: docs/source/en/main_classes/trainer.md (hunk 2 above)
Comment (stevhliu) [reply docs/source/en/main_classes/trainer.md:576]: I believe its still true, I didn't see `mps` among the supported backends for [`torch.distributed`](https://pytorch.org/docs/stable/distributed.html) (included in the second to last paragraph of the new doc)

File: docs/source/en/perf_train_special.md (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/perf_train_special.md:22]: I think you can set `use_cpu=True` here, but cc'ing @pacman100 who'll know more about it 🙂 

https://github.com/huggingface/transformers/blob/ffa04def0e6765cdf1d9518546506325a72e2bbe/src/transformers/training_args.py#L864

**Pull request merged**